components/script/dom/fontfaceset.rs

```diff
@@ -44,9 +52,20 @@ impl FontFaceSet {
         )
     }
 
+    pub(super) fn handle_font_face_load_completed(&self, font_face: &FontFace) {
+        if font_face.loaded() {
```

Comment:
Can you move this check to the `Add` method? It's a little weird this can be called for unloaded fonts. Then this could be an assertion.

Reply:
Eventually this method would handle unloaded fonts as well, to replace its promise and reset the status etc. Maybe I can just rename this to `handle_font_face_status_change`?